src/pkg/storage/storage.go

```diff
@@ -92,3 +93,79 @@ func orEmptyString(v any) string {
 
 	return v.(string)
 }
+
+func (s Storage) StorageConfig() (Configurer, error) {
+	switch s.Provider {
+	case ProviderS3:
+		return buildS3ConfigFromMap(s.Config)
+	}
+
+	return nil, clues.New("unsupported storage provider: " + s.Provider.String())
+}
+
+func NewStorageConfig(provider ProviderType) (Configurer, error) {
+	switch provider {
+	case ProviderS3:
+		return &S3Config{}, nil
+	}
+
+	return nil, clues.New("unsupported storage provider: " + provider.String())
+}
+
+type KVStoreGetter interface {
```

Comment:
Nit: Simpler interface names are better.
```suggestion
type Getter interface {
```